torrent/tracker.go

```diff
@@ -206,19 +208,24 @@ func (self *Tracker) serveFile(w http.ResponseWriter, r *http.Request, file *Fil
 		// TODO: This could run infinitely in a case where the file is requested and deleted or
 		// replaced, so we keep checking a structure that never will get filled in since it's no
 		// longer active.
+		file.Lock.Lock()
```

Comment:
This seems simpler. Right now, it just serializes around the single read in the check. Even if we have 20,000 people requesting it, I can't see a scenario where this becomes a significant bottleneck. (My guess is that the time-delays caused by lock-waits for a single-line critical section are probably lost in network latency.)

You're the owner, so you can tell me what you want, and I'll do it!  :-). But my inclination, based on being burned before, is to always go with the simplest thing that works, and only go to a more complex solution when I have data to convince me that it's necessary.